cmd/podman/common/specgen.go

```diff
@@ -778,28 +778,38 @@ func parseThrottleIOPsDevices(iopsDevices []string) (map[string]specs.LinuxThrot
 	return td, nil
 }
 
-func parseSecrets(secrets []string) ([]string, map[string]string, error) {
+func parseSecrets(secrets []string) ([]specgen.Secret, map[string]string, error) {
 	secretParseError := errors.New("error parsing secret")
-	var mount []string
+	var mount []specgen.Secret
 	envs := make(map[string]string)
 	for _, val := range secrets {
+		// mount only tells if user has set an option that can only be used with mount secret type
+		mountOnly := false
 		source := ""
 		secretType := ""
 		target := ""
+		var uid, gid uint32
+		// default mode 444 octal = 292 decimal
+		var mode uint32 = 292
```

Comment:
@flouthoc, @rhatdan  this line should solve the permissions issue, it defaults the secret to 444, which is what docker does. The default UID and GID for a secret inside a container is still 0 though, even if the container is run using --user, because that's what docker does, but it still allows other users to read the secret. All paths leading up to the secret are 755.  Let me know if this looks okay/

Reply:
This should be in Common, since buildah will also need this correct?

Reply:
Buildah already has something like this, though I would argue it shouldn't be in the current common, since the secrets package in common only understands how to store secrets, and should be agnostic to how the secrets are used. I'm open to possibly have a secretsUtil package under the secrets package in common for functions like this parse function and a copy secret to dir function, but I think @vrothberg has a strong opinion on keeping common secrets usage-agnostic.

Reply:
Ok, I would say we should move these helper functions then.